### PR TITLE
[TAS-1796] 🎨 Save redirect url to preAuthRoute

### PR DESCRIPTION
**Pull request by AuroraHuang22**

### CODE REVIEW

File: pages/in/register/index.vue [hunk 1]
```diff
@@ -191,6 +191,16 @@ export default {
     },
   },
   async mounted() {
+    const { redirect } = this.$route.query;
+    if (redirect) {
```
Comment (williamchong) [inline pages/in/register/index.vue:195]: Can we reuse `tryPostLoginRedirect` ?

**Review by williamchong — CHANGES_REQUESTED** [review]

File: pages/in/register/index.vue [hunk 2]
```diff
@@ -191,6 +191,16 @@ export default {
...
+      const redirectUrl = new URL(decodeURIComponent(redirect));
```
[the remark this answers is not in the document]
Comment (AuroraHuang22) [reply pages/in/register/index.vue]: [🥅 Catch error when setting redirect url](https://github.com/likecoin/like-co/pull/1742/commits/c0bb793856df656d37f5dba6ce257cc8a19fb6f8)

File: pages/in/register/index.vue (hunk 1 above)
Comment (williamchong) [reply pages/in/register/index.vue:195]: But in `tryPostLoginRedirect ` we only allow `REDIRECT_WHITE_LIST` to be redirected

Comment (AuroraHuang22) [reply pages/in/register/index.vue:195]: Got it, let me see see how to reuse

Comment (williamchong) [reply pages/in/register/index.vue:195]: Or just implement the white list here

Comment (AuroraHuang22) [reply pages/in/register/index.vue]: [🥅 Validate redirect URL](https://github.com/likecoin/like-co/pull/1742/commits/e34ca0ce03f34f01196e2cb221e6896a239b572d)

File: util/client.js [hunk 3]
```diff
@@ -47,7 +46,7 @@ export function openURL(vue, url, name, specs, replace) {
 
 export function tryPostLoginRedirect({ route }) {
   const { redirect, is_popup: isPopup } = route.query;
-  if (redirect && REDIRECT_WHITE_LIST.indexOf(parse(redirect, {}).hostname) !== -1) {
+  if (redirect && !isValidRedirectUrl(redirect)) {
```
Comment (williamchong) [inline util/client.js:49]: Is the condition flipped? `isValidRedirectUrl` should be true

**Review by williamchong — CHANGES_REQUESTED** [review]

File: pages/in/register/index.vue [hunk 4]
```diff
@@ -191,6 +199,11 @@ export default {
     },
   },
   async mounted() {
+    const { redirect } = this.$route.query;
+    if (redirect && isValidRedirectUrl(redirect)) {
```
Comment (williamchong) [inline pages/in/register/index.vue:203]: Also we should not need to manually call `decodeURIComponent` if we get `redirect` from `this.$route.query`, please confirm

File: util/client.js (hunk 3 above)
Comment (AuroraHuang22) [reply util/client.js]: [7369587](https://github.com/likecoin/like-co/pull/1742/commits/736958708167406d0003c08e94e1da4acfbcccce)

File: pages/in/register/index.vue (hunk 4 above)
Comment (AuroraHuang22) [reply pages/in/register/index.vue:203]: 👌 Vue Router has already decoded the URL
[🎨 Remove decodeURIComponent](https://github.com/likecoin/like-co/pull/1742/commits/336037823fa3cb081a21cb1d5b8f78a7f5c99a15)

Comment (williamchong) [reply pages/in/register/index.vue:203]: Do we still need decode in parseRedirectUrl ?
https://github.com/likecoin/like-co/pull/1742/files#diff-b41ab5cb5e0ab92f72bc78e948a8acfbde2d60d68ce1831b7089bdd9c4bf6739R428

Comment (AuroraHuang22) [reply pages/in/register/index.vue:203]: [🎨 Remove decodeURIComponent](https://github.com/likecoin/like-co/pull/1742/commits/5cefd3ed3805d3538ac46cc42f3aadd189a788bd)

**Review by williamchong — APPROVED** [review]

**Pull request merged**